### DISCUSSION TITLE
iOS phone screen disabled when CarPlay in use?

### DISCUSSION THREAD

**Discussion by EmmEff** [osmandapp/OsmAnd#12147] [General]
Apologies in advance if this is the wrong venue…

I installed OsmAnd for iOS for the first time in a long time yesterday and was pleasantly surprised to see it being a very full-functioning app compared to the early releases. I was also quite surprised to see it was CarPlay enabled. Previous to this I had been an Android user of the app.

Hats off to the developers for making a great app as always!

I am curious why the decision was made to disable the phone screen when CarPlay is in use? Since most of the functionality in the app is unavailable and cannot be made available through CarPlay, it makes it completely unusable (ie. can’t start/stop track recording, enable/disable display of POIs, change settings), etc.

I want to use this app with on motorcycle (Honda Africa Twin) but as soon as I plug it in, 90% of the functionality is inaccessible.

Is this something that will be addressed and hopefully resolved in the future?

**Comment by vshcherb** [comment]
As we keep investigating this issue, it looks like an iOS Car Play limitations. So technically we can enable screen but that doesn't follow guidelines and application won't be approved for distribution (https://developer.apple.com/design/human-interface-guidelines/carplay/overview/introduction/).

**Comment by EmmEff** [comment]
For what it's worth, most, if not all, other CarPlay apps keep both screens active- Rever, Gaia GPS Pro, Waze, Google Maps, etc.

**Reply by vshcherb** [reply]
Sounds reasonable, so I think we're going to improve this situation.

**Reply by vshcherb** [reply]
We plan to address it with https://github.com/osmandapp/OsmAnd-iOS/issues/1214

**Reply by EmmEff** [reply]
Please be sure to include the ability to set map orientation and zoom levels on the CarPlay screen. The ability to start/stop track recording, and other non-CarPlay features will need to remain on the (active) phone screen as I don’t think these are supported in CarPlay.

**Reply by hadventurerider** [reply]
Hello the team, and 👏 app improve each time ! using 4.1.0.6 on my Africa twin; 
still not possible to have the app on both screens 😡
not possible to begin navigation on the phone and continue on the dashboard ; have to begin again when connected
crashing (and leave) when asking for navigation (track) on the CarPlay screen (offently) 
have to ask again for navigation track after each stop, no keep it in memory ... 
when navigation is working, not easy to see white arrows on light grey cartridge ; please black arrows or dark grey on the background cartridge ... 
sometimes I have upward direction of movement (my choice) and some times not; no changes in parameters, only stop and go the bike ... ok on the phone' screen

**Reply by vshcherb** [reply]
Wow too many things might be in response. Crashing is bad and need to be fixed.
We actually don't consider to make 2 screens usable now as a very priority thing, the basic path should work though. You start navigation on the phone and continue on the car, once car navigation is finished, phone becomes active again.

**Reply by hadventurerider** [reply]
waiting for these improvements ;)
